tests/testthat/test-Subsets.R

```diff
@@ -232,3 +232,36 @@ test_that("subset generation", {
   expect_true(all(cohortsGenerated$generationStatus == "SKIPPED"))
   unlink(recordKeepingFolder, recursive = TRUE)
 })
+
+test_that("Subset definition creation and retrieval with definitionId != 1", {
+  sampleCohorts <- CohortGenerator::createEmptyCohortDefinitionSet()
+  cohortJsonFiles <- list.files(path = system.file("testdata/name/cohorts", package = "CohortGenerator"), full.names = TRUE)
+  cohortJsonFileName <- cohortJsonFiles[1]
+  cohortName <- tools::file_path_sans_ext(basename(cohortJsonFileName))
+  cohortJson <- readChar(cohortJsonFileName, file.info(cohortJsonFileName)$size)
+  sampleCohorts <- rbind(sampleCohorts, data.frame(
+    cohortId = as.double(1),
+    cohortName = cohortName,
+    json = cohortJson,
+    sql = "",
+    stringsAsFactors = FALSE
+  ))
+  
+  
+  # Limit to male only
+  subsetDef2 <- CohortGenerator::createCohortSubsetDefinition(
+    name ="Male Only",
+    definitionId = 2,
+    subsetOperators = list(
+      CohortGenerator::createDemographicSubset(id = 4,
+                                               name = "Male",
+                                               gender = 8507)
+    )
+  )
+  
+  sampleCohortsWithSubsets <- sampleCohorts %>%
+    CohortGenerator::addCohortSubsetDefinition(subsetDef2)
```

Comment:
test cases that uses overwriteExisting = T/F are needed, ensuring that an id cannot be duplicated (so, expect_error) and that it is actually overwritten